elFinder.NetCore/Extensions/IDirectoryExtensions.cs

```diff
@@ -0,0 +1,88 @@
+﻿using elFinder.NetCore.Drivers;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace elFinder.NetCore.Extensions
+{
+    public static class IDirectoryExtensions
+    {
+        public static byte GetLockedFlag(this IDirectory directory, RootVolume volume)
```

Comment:
The idea for these Get...Flag methods is to use the below priority (high to low) to determine the flag value:
- Check RootVolume settings (IsReadOnly, IsLocked)
- Check if there is any specific NamedItemAttribute for current File/Directory 
- Check if there is any NamedItemAttribute for parents (recursive until root is met) of current File/Directory.
- Use the default ItemAttribute set in RootVolume